### PR TITLE
under/oversized stories counters

### PR DESCRIPTION
**Pull request by pgulley**
Straightforward

### CODE REVIEW

File: indexer/workers/parser.py [hunk 1]
```diff
@@ -40,8 +40,10 @@ def process_story(
         #     or let fail from repeated retries???
         try:
             mdd = mcmetadata.extract(link, html)
-        except mcmetadata.exceptions.BadContentError as e:
-            raise QuarantineException(getattr(e, "message", repr(e)))
+        except mcmetadata.exceptions.BadContentError:
+            self.incr(
+                "undersized-stories"
+            )  # _PG: Or "unparsed-story" label ('reason', 'too-short'), and add counters for no html and no link too?
```
Comment (philbudne) [inline indexer/workers/parser.py:46]: A counter with multiple labels is easier to deal with (with thing.* queries) than multiple counter names (so long as only one label gets incremented).

I'd vote for a simple "stories" counter with status=ok, (or if you did status=ok-METHOD you could remove the parsed-stories counter)

File: indexer/workers/fetcher/fetch_worker.py [hunk 2]
```diff
@@ -196,18 +194,9 @@ def main_loop(self) -> None:
                     logger.warn(
                         f"Story over {MAX_FETCHER_MSG_SIZE} limit: {story.rss_entry().link}, size: {len(story_dump)}"
                     )
-                    oversized_stories += 1
+                    self.incr("oversized-stories")
```
Comment (philbudne) [inline indexer/workers/fetcher/fetch_worker.py:197]: As below (wrote that comment first), a single "stories" counter with different status values would mean that when a new status is added, it appears in the graphs without human intervention.

If the test is done based on HTML size alone (ie; `FETCHER_MAX_HTML`; which is a reasonable proxy, and easier for muggles to understand), the test can be done in `scrapy_cb` (along with replacing the various asserts there with different status values).  Logging the URL is good: it means we can grep the logs to see what sources are responsible.

Comment (philbudne) [reply indexer/workers/fetcher/fetch_worker.py:197]: regarding asserts: https://github.com/mediacloud/story-indexer/issues/118 for log messages from an assertion failure

Comment (pgulley) [reply indexer/workers/fetcher/fetch_worker.py:197]: Good idea to move the check into scrapy_cb! With a single counter in the fetcher, I'm concerned that a "status=oversized" label would overwrite the "http_xxx" label for oversized stories. I suppose that we don't anticipate very many oversized stories ever, so something like "status=oversized_http_xxx" might do?

**Comment by philbudne** [comment]
> "status=oversized_http_xxx"

Hmm.  I thought the only HTTP status that returned valid data was 200!

The rss-fetcher doesn't accept anything other than 200:
https://github.com/mediacloud/rss-fetcher/blob/main/fetcher/tasks.py#L830
(304 happens if we've set conditions on when NOT to send an unchanged document)

Is there a way to tell an HTTP server we won't accept bodies above a
certain size that scrapy uses??

Even if so, I'd say http_xxx and oversized are fine as is; there will
always be different causes for rejection for similar reasons, and
trying to interpret and align data at the collection level will be, at
best incomplete, and possibly confusing.

I try to keep counter names short, so they don't get truncated when
viewed.

**Review by philbudne — APPROVED** [review]

**Comment by philbudne** [comment]
@pgulley is this ready for merge?

**Pull request merged**